Docs: add data passing in components doc (#256)
oam-dev/kubevela.io: docs/end-user/component-dependency-parameter.md

title:  Data Pass Between Components 
---

This section will introduce how to pass data between components.

## How to use

> If we want to apply a WordPress server with data stored in a MySQL database, we need to pass the MySQL address to WordPress.

Reviewer: ```suggestion
In the following we will apply a WordPress server with the MySQL address passed from a MySQL component.
```

Apply the following `Application`:

```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: wordpress-with-mysql
  namespace: default
spec:
  components:
    - name: mysql
      type: helm
      outputs:
        # the output is the mysql service address
        - name: mysql-svc

>           exportKey: output.metadata.name + ".default.svc.cluster.local"

Reviewer: Isn't this not a key but a value?

Reviewer 2: I think this demo not work, please confirm @FogDong

Reviewer: Talked to @FogDong . She will rectify it.

Author: The exportKey can be made up of key and value, the test has been passed locally.

Ref: https://github.com/oam-dev/kubevela/blob/master/pkg/controller/core.oam.dev/v1alpha2/application/generator_test.go#L103
